base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -287,7 +290,7 @@ where T: BlockchainBackend + 'static
                 BlockAddResult::OrphanBlock => false,
                 BlockAddResult::ChainReorg(_) => true,
             };
-            if propagate {
+            if propagate && *propagate_out {
```

Comment:
I wonder if the code readability can be improved here.

Reply:
Perhaps I could rename the first one to Internal_propagate_check